ui/src/app/job-list/job-list.component.ts

```diff
@@ -75,6 +81,18 @@ export class JobListComponent implements OnInit {
         this.handleError(event.error);
       }
     });
+
+    // set project ID (if any) and get display field info for list columns
+    const req = URLSearchParamsUtils.unpackURLSearchParams(this.route.snapshot.queryParams['q']);
+    this.projectId = req.extensions.projectId || '';
+    const savedColumnSettings = this.settingsService.getDisplayColumns(this.projectId);
+    let field:DisplayField;
+    this.capabilities.displayFields.forEach((df) => {
+      field = df;
+      field.primary = true;
+      field.primary = !(savedColumnSettings.length && !savedColumnSettings.includes(df.field));
```

Comment:
The double negation here is a little confusing, can you do:
```
field.primary = !savedColumnSettings.length || savedColumnSettings.includes(df.field)
```

Reply:
good idea!